src/framework/components/anim/controller.js

```diff
@@ -0,0 +1,638 @@
+                    for (i = 0; i < state.animations.length; i++) {
+                        animation = state.animations[i];
+                        this._animEvaluator.findClip(animation.name).blendWeight = animation.weight / state.totalWeight;
+                    }
+                } else {
+                    var interpolatedTime = this._currTransitionTime / this._totalTransitionTime;
+
+                    for (i = 0; i < this._transitionPreviousStates.length; i++) {
+                        state = this._findState(this._transitionPreviousStates[i].name);
+                        var stateWeight = this._transitionPreviousStates[i].weight;
+                        for (j = 0; j < state.animations.length; j++) {
+                            animation = state.animations[j];
+                            this._animEvaluator.findClip(animation.name).blendWeight = (1.0 - interpolatedTime) * animation.weight / state.totalWeight * stateWeight;
+                        }
+                    }
+                    state = this.activeState;
+                    for (i = 0; i < state.animations.length; i++) {
+                        animation = state.animations[i];
+                        this._animEvaluator.findClip(animation.name).blendWeight = interpolatedTime * animation.weight / state.totalWeight;
+                    }
+
+                }
+                this._currTransitionTime += dt;
+            }
+            this._animEvaluator.update(dt);
+        },
+
+        findParameter: function (name) {
+            return this._parameters[name];
+        },
+
+        getParameterValue: function (name, type) {
+            var param = this.findParameter(name);
+            if (param && param.type === type) {
+                return param.value;
+            }
+            // #ifdef DEBUG
+            console.log('Cannot get parameter value. No parameter found in anim controller named "' + name + '" of type "' + type + '"');
+            // #endif
+        },
+
+        setParameterValue: function (name, type, value) {
+            var param = this.findParameter(name);
+            if (param && param.type === type) {
+                param.value = value;
+                return;
+            }
+            // #ifdef DEBUG
+            console.log('Cannot set parameter value. No parameter found in anim controller named "' + name + '" of type "' + type + '"');
+            // #endif
+        }
+    });
+
+    return {
+        ANIM_INTERRUPTION_SOURCE_NONE: ANIM_INTERRUPTION_SOURCE_NONE,
+        ANIM_INTERRUPTION_SOURCE_PREV_STATE: ANIM_INTERRUPTION_SOURCE_PREV_STATE,
+        ANIM_INTERRUPTION_SOURCE_NEXT_STATE: ANIM_INTERRUPTION_SOURCE_NEXT_STATE,
+        ANIM_INTERRUPTION_SOURCE_PREV_STATE_NEXT_STATE: ANIM_INTERRUPTION_SOURCE_PREV_STATE_NEXT_STATE,
+        ANIM_INTERRUPTION_SOURCE_NEXT_STATE_PREV_STATE: ANIM_INTERRUPTION_SOURCE_NEXT_STATE_PREV_STATE,
+
+        ANIM_TRANSITION_PREDICATE_GREATER_THAN: ANIM_TRANSITION_PREDICATE_GREATER_THAN,
```

Comment:
These constants seem crayzay long. Imagine typing all that out!

@willeastcott has spoken about using simple strings in some cases. Could this be such a case?

Otherwise perhaps we could loose the _SOURCE and _PREDICATE bits?

Reply:
Yeah these got way too long, i've shorted them.